x-pack/plugins/enterprise_search/common/ml_inference_pipeline/index.ts

```diff
@@ -40,11 +40,10 @@ export const LANG_IDENT_MODEL_TYPE = 'lang_ident';
 
 export interface MlInferencePipelineParams {
   description?: string;
-  destinationField: string;
+  fieldMappings: Record<string, string>;
```

Comment:
@sphilipse I'm not sure I understand. At this point each mapping in this object will have the key and value populated. If the user doesn't specify `destinationField`, it will get [set to a default](https://github.com/elastic/kibana/blob/main/x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/search_index/pipelines/ml_inference/ml_inference_logic.ts#L375). IMO it's safer to enforce a `string` value rather than allowing `undefined` and having to handle it in this code.

Reply:
So, the typing you've defined now means that TypeScript thinks that `fieldMappings` looks like this:
```
{
  a: string;
  aa: string;
  aaa: string;
  aaaa: string;
  ....
  zzzzz: string;
}
```
That is, this TypeScript definition means that every single possible key an object could ever have is treated as defined. Now: that's not necessarily a big problem here right now because you're only accessing it by looping over the defined keys, but it can lead to issues down the road.

Reply:
Ok, I changed the type to `Record<string, string | undefined>` and added logic to set a default when the value of a key is `undefined`.